vortex-scalar/src/binary.rs

```diff
@@ -57,3 +57,14 @@ impl<'a> TryFrom<&'a Scalar> for Buffer {
             .ok_or_else(|| vortex_err!("Can't extract present value from null scalar"))
     }
 }
+
+impl<'a> TryFrom<&'a Scalar> for bytes::Bytes {
+    type Error = VortexError;
+
+    fn try_from(value: &'a Scalar) -> VortexResult<Self> {
+        match Buffer::try_from(value)? {
+            Buffer::Arrow(b) => Ok(bytes::Bytes::from(b.as_slice().to_vec())),
```

Comment:
i always get a little uneasy when TryFrom/From can lead to silent allocations

Reply:
Agreed. Removed.

I feel a bit weird about not being able to round-trip a bytes. Maybe I should keep it but fail the Arrow case?

I'm still quite new to Rust and am not certain of the vibes of TryFrom. I can't find any explicit verbiage about allocation in the Rust docs for `From` or `TryFrom`. This does exist: [`impl<T, const N: usize> From<&[T; N]> for Vec<T>`](https://doc.rust-lang.org/std/convert/trait.From.html#impl-From%3C%26%5BT;+N%5D%3E-for-Vec%3CT%3E).

I guess the question is: does the broader Rust community or us the Spiral Rust community have a clear semantics for From/TryFrom wrt allocation?

Reply:
On the From-side, I'm not really sure.

On the Into-side, Rust does have some [recommendations](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv):

<img width="1071" alt="image" src="https://github.com/user-attachments/assets/d90abbdc-05ca-412a-afe6-20abdf1af382">

The difference is that calling `something.into_vec()` is explicit, whereas calling `something.into()` and dispatching to its `From` implementation which silently does heap allocations can be sneaky

Reply:
I think rust is a bit loose with it's allocations this is purely what kind of footgun do you expose. Since you know you're getting an owned value out of reference you might be aware that you will incur a copy. I think it's just a matter of how much you have to think in order to do the conversion. We have the same thing with strings but I think that one is mostly fine though falls into the same category of allocating froms as this